spec/requests/firmware_registries_spec.rb

```diff
@@ -0,0 +1,157 @@
+RSpec.describe "firmware_registries API" do
+  let!(:registry) { FactoryBot.create(:firmware_registry) }
+  let(:registry2) { FactoryBot.create(:firmware_registry) }
+
+  describe 'GET /api/firmware_registries' do
+    let(:url) { api_firmware_registries_url }
+
+    it 'returns the firmware_registries with an appropriate role' do
+      api_basic_authorize action_identifier(:firmware_registries, :read, :collection_actions, :get)
+      get(url)
```

Comment:
preferring to use the rails helper here i.e. get(api_firmware_registries_url) for readability and skip the url definition.  can be done in followup-pr.